demos/http/http_demo_mutual_tls/http_demo_mutual_tls.c

```diff
@@ -0,0 +1,351 @@
+                                      &response,
+                                      0 );
+    }
+    else
+    {
+        LogError( ( "Failed to initialize HTTP request headers: Error=%s.",
+                    HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %.*s%.*s...\n"
+                   "Response Headers:\n%.*s\n"
+                   "Response Status:\n%u\n"
+                   "Response Body:\n%.*s\n",
+                   ( int32_t ) IOT_CORE_ENDPOINT_LENGTH, IOT_CORE_ENDPOINT,
+                   ( int32_t ) pathLen, pPath,
+                   ( int32_t ) response.headersLen, response.pHeaders,
+                   response.statusCode,
+                   ( int32_t ) response.bodyLen, response.pBody ) );
+    }
+    else
+    {
+        LogError( ( "Failed to send HTTP %.*s request to %.*s%.*s: Error=%s.",
+                    ( int32_t ) methodLen, pMethod,
+                    ( int32_t ) IOT_CORE_ENDPOINT_LENGTH, IOT_CORE_ENDPOINT,
+                    ( int32_t ) pathLen, pPath,
+                    HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus != HTTP_SUCCESS )
+    {
+        returnStatus = EXIT_FAILURE;
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ *
+ * This example resolves the AWS IoT Core endpoint, establishes a TCP connection,
+ * performs a mutually authenticated TLS handshake occurs such that all further
+ * communication is encrypted. After which, HTTP Client Library API is used to
+ * make a POST request to AWS IoT Core in order to publish a message to a topic
+ * named topic with QoS=1 so that all clients subscribed to the topic receive
+ * the message at least once. Any possible errors are also logged.
+ *
+ * @note This example is single-threaded and uses statically allocated memory.
+ *
+ */
+int main( int argc,
+          char ** argv )
+{
+    /* Return value of main. */
+    int returnStatus = EXIT_SUCCESS;
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t transportInterface;
+    /* The network context for the transport layer interface. */
+    NetworkContext_t networkContext;
+    /* Credentials to establish the TLS connection. */
+    OpensslCredentials_t opensslCredentials;
+    /* Status returned by OpenSSL transport implementation. */
+    OpensslStatus_t opensslStatus;
+    /* Information about the server to send the HTTP request. */
+    ServerInfo_t serverInfo;
+
+    ( void ) argc;
+    ( void ) argv;
+
+    /* Initialize TLS credentials. */
+    ( void ) memset( &opensslCredentials, 0, sizeof( opensslCredentials ) );
+    opensslCredentials.pClientCertPath = CLIENT_CERT_PATH;
+    opensslCredentials.pPrivateKeyPath = CLIENT_PRIVATE_KEY_PATH;
+    opensslCredentials.pRootCaPath = ROOT_CA_CERT_PATH;
```

Comment:
We are expecting these certificates to always be NULL terminated?

Reply:
As discussed offline, this is done because the underlying API call requires NULL terminated strings.